src/dnx.coreclr/dnx.coreclr.cpp

```diff
@@ -35,344 +27,185 @@ bool GetTrustedPlatformAssembliesList(const wchar_t* szDirectory, bool bNative,
 
-Finished:
-    return hCoreCLRModule;
+    return LoadLibraryExW(dnx::utils::path_combine(coreclr_dir, L"coreclr.dll").c_str(), NULL, 0);
 }
 
-
-/*
-    Win2KDisable : DisallowWin32kSystemCalls
-    SET DNX_WIN32K_DISABLE=1
-*/
-
-bool Win32KDisable()
+bool PinModule(HMODULE module, dnx::trace_writer& trace_writer)
 {
-    bool fSuccess = true;
-    TCHAR szWin32KDisable[2] = {};
-    LPWSTR lpwszModuleFileName = L"api-ms-win-core-processthreads-l1-1-1.dll";
-    HMODULE hProcessThreadsModule = nullptr;
-    // Note: Need to keep as ASCII as GetProcAddress function takes ASCII params
-    LPCSTR pszSetProcessMitigationPolicy = "SetProcessMitigationPolicy";
-    FnSetProcessMitigationPolicy pFnSetProcessMitigationPolicy = nullptr;
-    PROCESS_MITIGATION_SYSTEM_CALL_DISABLE_POLICY systemCallDisablePolicy = {};
-    systemCallDisablePolicy.DisallowWin32kSystemCalls = 1;
-
-    DWORD dwRet = GetEnvironmentVariableW(L"DNX_WIN32K_DISABLE", szWin32KDisable, _countof(szWin32KDisable));
-    fSuccess = dwRet > 0;
-    if (!fSuccess)
-    {
-        goto Finished;
-    }
+    wchar_t module_path_buffer[MAX_PATH];
+    GetModuleFileName(module, module_path_buffer, MAX_PATH);
 
-    if (wcscmp(szWin32KDisable, L"1") != 0)
+    HMODULE ignoreModule;
+    // Pin the module - CoreCLR.dll does not support being unloaded.
+    if (!GetModuleHandleExW(GET_MODULE_HANDLE_EX_FLAG_PIN, module_path_buffer, &ignoreModule))
     {
-        fSuccess = false;
-        goto Finished;
+        trace_writer.write(L"Failed to pin coreclr.dll", false);
+        return false;
     }
 
-    fSuccess = KlrLoadLibraryExWAndGetProcAddress(
-                    lpwszModuleFileName,
-                    pszSetProcessMitigationPolicy,
-                    &hProcessThreadsModule,
-                    (FARPROC*)&pFnSetProcessMitigationPolicy);
-    if (!fSuccess)
-    {
-        goto Finished;
-    }
+    return true;
+}
 
-    if (!hProcessThreadsModule || !pFnSetProcessMitigationPolicy)
-    {
-        fSuccess = false;
-        goto Finished;
-    }
+HMODULE LoadCoreClr(dnx::trace_writer& trace_writer)
+{
+    HMODULE coreclr_module;
 
-    if (pFnSetProcessMitigationPolicy(
-              ProcessSystemCallDisablePolicy,   //_In_  PROCESS_MITIGATION_POLICY MitigationPolicy,
-              &systemCallDisablePolicy,         //_In_  PVOID lpBuffer,
-              sizeof(systemCallDisablePolicy)  //_In_  SIZE_T dwLength
-            ))
+    wchar_t coreclr_dir_buffer[MAX_PATH];
+    if (GetEnvironmentVariableW(L"CORECLR_DIR", coreclr_dir_buffer, MAX_PATH))
     {
-        printf_s("DNX_WIN32K_DISABLE successful.\n");
+        coreclr_module = LoadCoreClr(coreclr_dir_buffer, trace_writer);
     }
 
-Finished:
-    //Cleanup
-    if (pFnSetProcessMitigationPolicy)
-    {
-        pFnSetProcessMitigationPolicy = nullptr;
-    }
+    coreclr_module = LoadLibraryExW(L"coreclr.dll", NULL, 0);
```

Comment:
I used to have a return from inside the `if` block but I decided to treat pining as part of loading and did not change the code correctly